recipes/mbits-args/all/conanfile.py

```diff
@@ -1,103 +1,127 @@
-        elif tools.Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration("mbits-args: Unsupported compiler: {} {}; "
-                                            "minimal version known to work is {}."
-                                            .format(self.settings.compiler, self.settings.compiler.version, minimum_version))
-        elif str(self.settings.compiler) == "clang" and tools.Version(self.settings.compiler.version) < "8":
-            libcxx = self.settings.compiler.get_safe("libcxx")
-            if libcxx and str(libcxx) == "libc++":
-                raise ConanInvalidConfiguration("mbits-args: Unsupported compiler: clang {} with libc++;\n"
-                                                "minimal version known to work is either clang 8 with "
-                                                "libc++ or clang {} with libstdc++/libstdc++11."
-                                                .format(self.settings.compiler.version, minimum_version))
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("args-{}".format(self.version),
-                  self._source_subfolder)
-
-    def _configure_cmake(self):
-        if self._cmake is not None:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["LIBARGS_TESTING"] = False
-        self._cmake.definitions["LIBARGS_INSTALL"] = True
-        self._cmake.definitions["LIBARGS_SHARED"] = self.options.shared
-        self._cmake.configure()
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LIBARGS_TESTING"] = False
+        tc.variables["LIBARGS_INSTALL"] = True
+        tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        copy(
+            self,
+            pattern="LICENSE",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        cmake = CMake(self)
         cmake.install()
-        self.copy("LICENSE",
-                  "licenses", keep_path=False, src=self._source_subfolder)
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
```

Comment:
There's are all from the template but I am pretty sure we could remove them. I dont see these files in the logs

Reply:
I suppose .pdb might be created on Windows, will double check